src/wp-admin/network/users.php

```diff
@@ -91,11 +91,22 @@
 								}
 
 								$userfunction = 'all_spam';
-								$blogs        = get_blogs_of_user( $user_id, true );
 
-								foreach ( (array) $blogs as $details ) {
-									if ( ! is_main_site( $details->userblog_id ) ) { // Main site is not a spam!
-										update_blog_status( $details->userblog_id, 'spam', '1' );
+								/**
+								 * Filters whether to propagate the blog status when a user is marked as spam.
+								 *
+								 * @since 6.9.0
+								 *
+								 * @param bool $propagate Whether to propagate the blog status. Default false.
+								 * @param int  $user_id   User ID.
+								 */
+								if ( apply_filters( 'network_user_spam_propagate_to_blogs', false, $user_id, $doaction ) ) {
```

Comment:
The `$doaction` param can be removed:

```suggestion
								if ( apply_filters( 'network_user_spam_propagate_to_blogs', false, $user_id ) ) {
```